### PR TITLE
Update the History.txt section for V4.0.0 Release

### PR DESCRIPTION
**Pull request by shubnil**
Update the history section for V4.0.0 release.

Description
-----------
This captures the various changes  and documentation updates and felicitations to the various contributors.

Test Steps
-----------

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] I have modified and/or added unit-tests to cover the code changes in this Pull Request.

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by ActoryOu — DISMISSED** [review]

File: History.txt [hunk 1]
```diff
@@ -1,3 +1,53 @@
...
+	+ Stack is Penetration tested and a memory leak issue has been identified and fixed in the process.
+	(Refrence: https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/973)
+	+ Improved Documentation:
+		1. Multi-interface and Multi-endpoint Documentataion.
+		2. Updated stack initialization documentation to support multi-interface and endpoint.
+		(Reference: https://freertos-wordpress.corp.amazon.com/FreeRTOS-Plus/FreeRTOS_Plus_TCP/TCP_Networking_Tutorial_Initialising_TCP.updated.html)
+		3. Code size details for various build configurations
+		(Reference: https://www.freertos.org/FreeRTOS-Plus/FreeRTOS_Plus_TCP/index.html).
+		4. Documentataion added for IPv6 protocol support.
+		5. Documentataion added for various IPv6 functionalities like DHCPv6, ND (Neighbour Discovery), RA (Router Advertisement).
```
Comment (sirnish) [inline History.txt:32]: Documentataion => Documentation

File: History.txt [hunk 2]
```diff
@@ -1,3 +1,53 @@
...
+	+ Memory safety proofs for IPv6 code using[ CBMC automated reasoning tool ] ( https:/*www.cprover.org/cbmc/).
+	+ Stack is Penetration tested and a memory leak issue has been identified and fixed in the process.
+	(Refrence: https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/973)
+	+ Improved Documentation:
+		1. Multi-interface and Multi-endpoint Documentataion.
+		2. Updated stack initialization documentation to support multi-interface and endpoint.
+		(Reference: https://freertos-wordpress.corp.amazon.com/FreeRTOS-Plus/FreeRTOS_Plus_TCP/TCP_Networking_Tutorial_Initialising_TCP.updated.html)
+		3. Code size details for various build configurations
+		(Reference: https://www.freertos.org/FreeRTOS-Plus/FreeRTOS_Plus_TCP/index.html).
+		4. Documentataion added for IPv6 protocol support.
```
Comment (sirnish) [inline History.txt:31]: Documentataion => Documentation

File: History.txt [hunk 3]
```diff
@@ -1,3 +1,53 @@
+Documentation and download available at https://www.FreeRTOS.org/
+
+Changes between FreeRTOS-plus-TCP V4.0.0 and V3.1.0 released August 10, 2023
+	+ Unified code for both IPv4 and IPv6 protocols.
+        + Multiple Interface and Multiple Endpoint support
+		1. The stack supports multiple Physical Interfaces.
+		2. The stack also supports multiple endpoints (or Virtual interfaces) on the same physiscal interface.
```
Comment (sirnish) [inline History.txt:7]: physiscal => physical

File: History.txt (hunk 2 above)
Comment (shubnil) [reply History.txt:31]: Updated

File: History.txt (hunk 3 above)
Comment (shubnil) [reply History.txt]: Done

File: History.txt [hunk 4]
```diff
@@ -1,3 +1,53 @@
+Documentation and download available at https://www.FreeRTOS.org/
+
+Changes between FreeRTOS-plus-TCP V4.0.0 and V3.1.0 released August 10, 2023
+	+ Unified code for both IPv4 and IPv6 protocols.
+        + Multiple Interface and Multiple Endpoint support
+		1. The stack supports multiple Physical Interfaces.
+		2. The stack also supports multiple endpoints (or Virtual interfaces) on the same physical interface.
+	(Reference: https://www.freertos.org/FreeRTOS-Plus/FreeRTOS_Plus_TCP/freertostcp-multiple-interfaces.html).
```
Comment (moninom1) [inline History.txt:8]: This link will be updated with website. Can we skip the link update for now?

File: History.txt [hunk 5]
```diff
@@ -1,3 +1,53 @@
...
+		    2. IPv4 + UDP + TCP
+		    3. IPv6 + UDP
+		    4. IPv6 + UDP + TCP
+		    5. IPv4 + IPv6 + UDP
+		    6. IPv4 + IPv6 + UDP + TCP
+	+ Support added in FreeRTOS-plus-TCP git repository CI tool to validate various build configurations before PR merge.
+	+ Achieved 100% unit test coverage for both IPv4 and IPv6.
+	+ MISRA c-2012 compliance for both IPv4 and IPv6.
+	+ IPv6 code is protocol tested using Maxwell protocol tester.
+	+ Memory safety proofs for IPv6 code using[ CBMC automated reasoning tool ] ( https:/*www.cprover.org/cbmc/).
```
Comment (moninom1) [inline History.txt:22]: https:/*www.cprover.org/cbmc/ - Can the * be removed?

Comment (moninom1) [reply History.txt:22]: txt file does not anyway support links so it think [text](link) - this format will not work

File: History.txt (hunk 4 above)
Comment (shubnil) [reply History.txt:8]: We will update once link is updated

File: History.txt (hunk 5 above)
Comment (shubnil) [reply History.txt:22]: Any reason you feel that it is not needed

Comment (shubnil) [reply History.txt:22]: I think you meant "*". Got it. Let me update

Comment (shubnil) [reply History.txt]: Done

File: History.txt (hunk 4 above)
Comment (shubnil) [reply History.txt]: Removed the link. I will create a new PR with the new link

**Review by tony-josi-aws — APPROVED** [review]

File: History.txt [hunk 6]
```diff
@@ -1,3 +1,52 @@
...
+		    1. IPv4 + UDP
+		    2. IPv4 + UDP + TCP
+		    3. IPv6 + UDP
+		    4. IPv6 + UDP + TCP
+		    5. IPv4 + IPv6 + UDP
+		    6. IPv4 + IPv6 + UDP + TCP
+	+ Support added in FreeRTOS-plus-TCP git repository CI tool to validate various build configurations before PR merge.
+	+ Achieved 100% unit test coverage for both IPv4 and IPv6.
+	+ MISRA c-2012 compliance for both IPv4 and IPv6.
+	+ IPv6 code is protocol tested using Maxwell protocol tester.
```
Comment (amazonKamath) [inline History.txt:20]: Please change this to addressed several issues observed by Maxwell Protocol tester instead.

**Review by amazonKamath — APPROVED** [review]

File: History.txt (hunk 6 above)
Comment (shubnil) [reply History.txt]: Done

**Review by amazonKamath — APPROVED** [review]

**Review by sirnish — APPROVED** [review]
GTG

**Review by moninom1 — APPROVED** [review]

**Review by tony-josi-aws — APPROVED** [review]

**Review by moninom1 — APPROVED** [review]

**Pull request merged**